docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md

```diff
@@ -1,5 +1,30 @@
 # This document lists known breaking changes in Roslyn after .NET 6 all the way to .NET 7.
 
+## Nameof operator in attribute on method or local function
+
+***Introduced in .NET SDK 7.0.400, Visual Studio 2022 version 17.3.***
+
+When the language version is C# 11 or later, a `nameof` operator in an attribute on a method
+brings the type parameters of that method in scope. The same applies for local functions.
+
+For instance, this will now be an error:
+```csharp
+class C
+{
+  class TParameter
+  {
+    const string Constant = """";
+  }
+  [MyAttribute(nameof(TParameter.Constant))]
+  void M<TParameter>() { }
+}
+```
+
+Possible workarounds are:
+
+1. Rename the type parameter to avoid shadowing the name from outer scope.
```

Comment:
2 nits:
1. `Constant` is not accessible from the attribute, perhaps it should be made `public` to avoid confusion.
2. Qualifying the name of the constant also seems like an option here. e.g. `[MyAttribute(nameof(C.TParameter.Constant))]`.